app/components/avo/button_component.rb

```diff
@@ -1,9 +1,9 @@
 # frozen_string_literal: true
 
-# A button/link can have the following settings:
-# style: primary/outline/text
-# size: :xs :sm, :md, :lg
 class Avo::ButtonComponent < Avo::BaseComponent
+  SIZE = _Union(:xs, :sm, :md, :lg, :xl)
+  STYLE = _Union(:primary, :outline, :text, :icon)
```

Comment:
This seemed like the right place to define these unions, even though I’m not updating the `ButtonComponent` just yet.

I tried to find a limited set of colors, but it looks like these are just interpolated directly into the class list. That brings up an interesting concern: how does the Tailwind JIT know which colors to build? Have we listed them somewhere else so the JIT knows?

Reply:
> how does the Tailwind JIT know which colors to build? Have we listed them somewhere else so the JIT knows?

There are some standard ones, but it's also possible to do a [tailwind integration](https://docs.avohq.io/3.0/tailwindcss-integration.html) and use custom colors. Let's not define any restriction on colors